Add the huggingface token parameter, and modify the file path in llam…

melodyliulab opened the pull request:
I want to use the mistralai/Mistral-7B-Instruct-v0.2 models, and found there are no gguf files in HuggingFace, then I decided to use the ./convert_models functions to convert the model. I found there are some issues exist:

1.  **401 Client Error**

```
huggingface_hub.utils._errors.GatedRepoError: 401 Client Error. (Request ID: Root=1-66b1862a-1bc229376e7f3f4020a3c951;60195d59-03d1-4f26-b3ce-d3b04c2fe2b4)
Cannot access gated repo for url https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.2/resolve/36d7e540e651b68dac59394d9c3381651df7fb01/.gitattributes
```
So I added the HF_TOKEN=<YOUR_HF_TOKEN_ID> parameter in the code. 
Impacted files: README.md, download_huggingface.py, run.sh

2. **No convert.py and quantize files under llama.cpp**

```
python: can't open file '/opt/app-root/src/converter/llama.cpp/convert.py': [Errno 2] No such file or directory
run.sh: line 23: llama.cpp/quantize: No such file or directory
```

If we go to https://github.com/ggerganov/llama.cpp.git, we can find the convert.py has been deprecated and moved to examples/convert_legacy_llama.py. I am not sure if I should just keep the line  "python llama.cpp/convert-hf-to-gguf.py /opt/app-root/src/converter/converted_models/$hf_model_url", I just replace the convert.py with the correct path. also for llama.cpp/quantize

Impacted file: run.sh

3. **No image name was specified in the README.md**

So I added "localhost/converter" in the "podman run" command.


**Here is my testing after the modification**:
```
$ podman run -it --rm -v models:/converter/converted_models -e HF_MODEL_URL=mistralai/Mistral-7B-Instruct-v0.2 -e HF_TOKEN=*** -e QUANTIZATION=Q4_K_M -e KEEP_ORIGINAL_MODEL="False" localhost/converter

README.md: 100%|███████████████████████████████████████████████████████████████████████████████████| 5.47k/5.47k [00:00<00:00, 21.9MB/s]
.gitattributes: 100%|██████████████████████████████████████████████████████████████████████████████| 1.52k/1.52k [00:00<00:00, 8.79MB/s]
model.safetensors.index.json: 100%|█████████████████████████████████████████████████████████████████| 25.1k/25.1k [00:00<00:00, 357kB/s]
config.json: 100%|█████████████████████████████████████████████████████████████████████████████████████| 596/596 [00:00<00:00, 3.67MB/s]
generation_config.json: 100%|███████████████████████████████████████████████████████████████████████████| 111/111 [00:00<00:00, 621kB/s]
pytorch_model.bin.index.json: 100%|████████████████████████████████████████████████████████████████| 23.9k/23.9k [00:00<00:00, 72.1MB/s]
special_tokens_map.json: 100%|█████████████████████████████████████████████████████████████████████████| 414/414 [00:00<00:00, 6.70MB/s]
tokenizer.model: 100%|████████████████████████████████████████████████████████████████████████████████| 493k/493k [00:00<00:00, 861kB/s]
tokenizer_config.json: 100%|███████████████████████████████████████████████████████████████████████| 2.10k/2.10k [00:00<00:00, 12.7MB/s]
tokenizer.json: 100%|███████████████████████████████████████████████████████████████████████████████| 1.80M/1.80M [00:02<00:00, 630kB/s]
model-00001-of-00003.safetensors: 100%|████████████████████████████████████████████████████████████| 4.94G/4.94G [52:42<00:00, 1.56MB/s]
model-00003-of-00003.safetensors: 100%|██████████████████████████████████████████████████████████| 4.54G/4.54G [1:01:03<00:00, 1.24MB/s]
pytorch_model-00001-of-00003.bin: 100%|██████████████████████████████████████████████████████████| 4.94G/4.94G [1:05:53<00:00, 1.25MB/s]
pytorch_model-00002-of-00003.bin: 100%|██████████████████████████████████████████████████████████| 5.00G/5.00G [1:06:22<00:00, 1.26MB/s]
model-00002-of-00003.safetensors: 100%|██████████████████████████████████████████████████████████| 5.00G/5.00G [1:07:19<00:00, 1.24MB/s]
pytorch_model-00003-of-00003.bin: 100%|██████████████████████████████████████████████████████████| 5.06G/5.06G [1:07:36<00:00, 1.25MB/s]
Fetching 16 files: 100%|█████████
INFO:convert:Loading model file /opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2/model-00001-of-00003.safetensorsmodel-00002-of-00003.bin:  99%|██████████████████████████████████████████████████████████▍| 4.95G/5.00G [5:50:49<03:48, 222kB/s]
INFO:convert:Loading model file /opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2/model-00001-of-00003.safetensorsmodel-00002-of-00003.bin: 100%|███████████████████████████████████████████████████████████| 5.00G/5.00G [5:54:12<00:00, 229kB/s]
....
INFO:convert:Loading model file /opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2/model-00002-of-00003.safetensors
INFO:convert:Loading model file /opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2/model-00003-of-00003.safetensors
INFO:convert:params = Params(n_vocab=32000, n_embd=4096, n_layer=32, n_ctx=32768, n_ff=14336, n_head=32, n_head_kv=8, n_experts=None, n_experts_used=None, f_norm_eps=1e-05, rope_scaling_type=None, f_rope_freq_base=1000000.0, f_rope_scale=None, n_ctx_orig=None, rope_finetuned=None, ftype=None, path_model=PosixPath('/opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2'))
INFO:convert:Loaded vocab file PosixPath('/opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2/tokenizer.model'), type 'spm'
INFO:convert:model parameters count : (7241732096, 7241732096, 0) (7.2B)
INFO:convert:Vocab info: <SentencePieceVocab with 32000 base tokens and 0 added tokens>
INFO:convert:Special vocab info: <SpecialVocab with 0 merges, special tokens {'bos': 1, 'eos': 2, 'unk': 0}, add special tokens {'bos': True, 'eos': False}>
INFO:convert:Writing /opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2/Mistral-7B-Instruct-v0.2-F32.gguf, format 0
......
INFO:gguf.gguf_writer:Writing the following files:
INFO:gguf.gguf_writer:/opt/app-root/src/converter/converted_models/mistralai/Mistral-7B-Instruct-v0.2/Mistral-7B-Instruct-v0.2-F32.gguf: n_tensors = 291, total_size = 29.0G
INFO:convert:[  1/291] Writing tensor token_embd.weight                      | size  32000 x   4096  | type F32  | T+   0
INFO:convert:[  2/291] Writing tensor blk.0.attn_norm.weight                 | size   4096           | type F32  | T+   1
INFO:convert:[  3/291] Writing tensor blk.0.ffn_down.weight                  | size   4096 x  14336  | type F32  | T+   1
INFO:convert:[  4/291] Writing tensor blk.0.ffn_gate.weight                  | size  14336 x   4096  | type F32  | T+   1
....
```
(the log is too long)

fabiendupont committed:
Driver Toolkit: Get kernel version from bootc image

When building the `driver-toolkit` image, It is cumbersome to find kernel
version that matches the future `nvidia-bootc` and `intel-bootc` images.
However, the kernel version is stored as a label on the `rhel-bootc`
images, which are exposed as the `FROM` variable in the Makefile.

This change collects the kernel version using `skopeo inspect` and `jq`.

The `DRIVER_TOOLKIT_BASE_IMAGE` variable is introduced in the Makefile
to dissociate it from the `FROM` variable that is used as the `nvidia-bootc`
and `intel-bootc` base image.

The user can now specify something like:

```shell
make nvidia-bootc \
    FROM=quay.io/centos-bootc/centos-bootc:stream9 \
    DRIVER_TOOLKIT_BASE_IMAGE=quay.io/centos/centos:stream9
```

Also, the `VERSION` variable in `/etc/os-release` is the full version, so
this change modifies the command to retrieve the `OS_VERSION_MAJOR`
value.

Signed-off-by: Fabien Dupont <fdupont@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

melodyliulab committed:
Add the huggingface token parameter, and modify the file path in llama.cpp repo.

Signed-off-by: Song Liu <soliu@redhat.com>

fabiendupont committed:
Intel bootc - Update Gaudi drivers to 1.17.0-495

Intel has released the version `1.17.0-495` of their Gaudi drivers. They
are available explicitly for RHEL 9.4 with a new `9.4` folder in the RPM
repository. This change updates the arguments to use the new version
from the new repository folder.

Signed-off-by: Fabien Dupont <fdupont@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

melodyliulab committed:
Update run.sh

also the file name changed to "_" in https://github.com/ggerganov/llama.cpp, so change the file name from llama.cpp/convert-hf-to-gguf.py to llama.cpp/convert_hf_to_gguf.py

Signed-off-by: Song Liu <soliu@redhat.com>

melodyliulab committed:
Update run.sh

From the https://github.com/ggerganov/llama.cpp/blob/master/Makefile, it said "The 'quantize' binary is deprecated. Please use 'llama-quantize' instead."

The command works after my testing using llama-quantize.

Signed-off-by: Song Liu <soliu@redhat.com>

lstocchi committed:
fix: fix image name on chatbot ai-lab.yaml

Signed-off-by: lstocchi <lstocchi@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

javipolo committed:
RHELAI-838 Enforce insights registration in ilab wrapper~

Signed-off-by: Javi Polo <jpolo@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

pastequo committed:
feat(nvidia-bootc): Retag instructlab image if INSTRUCTLAB_IMAGE_RETAG is specified

Signed-off-by: Matthieu Bernardin <mbernard@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

axel7083 committed:
fix(whispercpp): set app ownership to user 1001

Signed-off-by: axel7083 <42176370+axel7083@users.noreply.github.com>
Signed-off-by: Song Liu <soliu@redhat.com>

axel7083 committed:
fix(whisper_cpp): missing ownership

Signed-off-by: axel7083 <42176370+axel7083@users.noreply.github.com>
Signed-off-by: Song Liu <soliu@redhat.com>

rhatdan commented:
You need to sign commit
git commit -a --amend -s
git push --force

melodyliulab commented:
> You need to sign commit git commit -a --amend -s git push --force
Done, please check again.

axel7083 committed:
chore: tag model_servers images on release

Signed-off-by: axel7083 <42176370+axel7083@users.noreply.github.com>

fix: missing $

Signed-off-by: axel7083 <42176370+axel7083@users.noreply.github.com>
Signed-off-by: Song Liu <soliu@redhat.com>

rhatdan commented:
@MichaelClifford PTAL

rhatdan commented:
Do you always need an HF_TOKEN? Can this still be used without it?  If so can we make it optional in the podman run command?  (I could be sadly mistaken).

maysunfaisal committed:
Configure model for rag recipe

Signed-off-by: Maysun J Faisal <maysunaneek@gmail.com>
Signed-off-by: Song Liu <soliu@redhat.com>

fabiendupont committed:
NVIDIA - Add the nvidia-driver package

The `nvidia-driver` package provides the firmware files for the given
driver version. This change removes the copy of the firmware from the
builder step and install the `nvidia-driver` package instead. This also
allows a better tracability of the files in the final image.

Signed-off-by: Fabien Dupont <fdupont@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

melodyliulab commented:
> Do you always need an HF_TOKEN? Can this still be used without it? If so can we make it optional in the podman run command? (I could be sadly mistaken).

So how to do that? copy the HF_TOKEN into the image by Containerfile?

n1hility committed:
fix: resolve fork failures during training runs

torchrun jobs create a number of children per GPU which can
often exceed the 2k limit.

Signed-off-by: Jason T. Greene <jason.greene@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

axel7083 committed:
chore(summarizer): adding markdown type file input

Signed-off-by: axel7083 <42176370+axel7083@users.noreply.github.com>
Signed-off-by: Song Liu <soliu@redhat.com>

axel7083 committed:
fix: file type check

Signed-off-by: axel7083 <42176370+axel7083@users.noreply.github.com>
Signed-off-by: Song Liu <soliu@redhat.com>

javipolo committed:
RHELAI-869: Hardcode XDG_RUNTIME_DIR path in case the variable does not exist

Signed-off-by: Javi Polo <jpolo@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

rhatdan commented:
No I am questioning whether this change forces users to always have and specify a token.  I don't really know how this all works, but it seems that if an image is available without a token, this change will force users to specify a token even if one does not exist.

MichaelClifford commented on convert_models/run.sh:
```diff
@@ -1,6 +1,7 @@
 #! /bin/bash
 
 hf_model_url=${HF_MODEL_URL}
+hf_token=${HF_TOKEN}
```
```suggestion
hf_token=${HF_TOKEN:="None"}
```
This should overcome the issue of when its not needed. 

Can you also add this token parameter as a textbox in the UI? otherwise UI users won't be able to access token gated models.

MichaelClifford commented on convert_models/README.md:
````diff
@@ -22,7 +22,7 @@ podman build -t converter .
 
 ```bash
-podman run -it --rm -v models:/converter/converted_models -e HF_MODEL_URL=<ORG/MODEL_NAME> -e QUANTIZATION=Q4_K_M -e KEEP_ORIGINAL_MODEL="False"
+podman run -it --rm -v models:/converter/converted_models -e HF_MODEL_URL=<ORG/MODEL_NAME> -e HF_TOKEN=<YOUR_HF_TOKEN_ID> -e QUANTIZATION=Q4_K_M -e KEEP_ORIGINAL_MODEL="False" localhost/converter
````
```suggestion
podman run -it --rm -v models:/converter/converted_models -e HF_MODEL_URL=<ORG/MODEL_NAME> -e HF_TOKEN=<YOUR_HF_TOKEN_ID> -e QUANTIZATION=Q4_K_M -e KEEP_ORIGINAL_MODEL="False" converter
```
I don't think you need the "localhost" prefix

melodyliulab replied:
I changed the following things according to your suggestions:

1. [convert_models/README.md]:
"HF_TOKEN" is optional, it is required for private models.
```bash
podman run -it --rm -v models:/converter/converted_models -e HF_MODEL_URL=<ORG/MODEL_NAME> -e HF_TOKEN=<YOUR_HF_TOKEN_ID> -e QUANTIZATION=Q4_K_M -e KEEP_ORIGINAL_MODEL="False" converter
```

2. run.sh
`hf_token=${HF_TOKEN:="None"}`

3. ui.py
```
token_id = st.text_input(label="Enter your huggingface token (optional)",
                         help="huggingface token is required for private model"
                        ) or "None"

                       "-v", f"{volume}:/converter/converted_models", 
                        "-e", f"HF_MODEL_URL={model_name}",
                        "-e", f"HF_TOKEN={token_id}",
```

Here is my testing:
![Screenshot at 2024-08-14 17-35-34](https://github.com/user-attachments/assets/e997425e-2018-4b5e-8f2d-9129a7d83262)

Not familiar with the PR workflow, thanks for the understanding.

javipolo committed:
Move --target-arch to image builder args

Signed-off-by: Javi Polo <jpolo@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

fabiendupont committed:
Modify condition on NVIDIA FabricManager service

The `/dev/nvswitchctl` device is created by the NVIDIA Fabric Manager
service, so it cannot be a condition for the `nvidia-fabricmanager`
service.

Looking at the NVIDIA driver startup script for Kubernetes, the actual
check is the presence of `/proc/driver/nvidia-nvswitch/devices` and the
fact that it's not empty [1].

This change modifies the condition to
`ConditionDirectoryNotEmpty=/proc/driver/nvidia-nvswitch/devices`, which
verifies that a certain path exists and is a non-empty directory.

[1] https://gitlab.com/nvidia/container-images/driver/-/blob/main/rhel9/nvidia-driver?ref_type=heads#L262-269

Signed-off-by: Fabien Dupont <fdupont@redhat.com>
Signed-off-by: Song Liu <soliu@redhat.com>

melodyliulab committed:
Update the HF_TOKEN to optional, and modify the UI.

Signed-off-by: Song Liu <soliu@redhat.com>

melodyliulab committed:
Also modify the run.sh and README.md to make the HF_TOKEN para optional.

Signed-off-by: Song Liu <soliu@redhat.com>

melodyliulab commented:
Made mistakes when sign-off, will fork a new branch to implement the changes.
